src/index/hnsw/faiss_hnsw.cc

```diff
@@ -140,6 +141,20 @@ class BaseFaissIndexNode : public IndexNode {
     AddInternal(const DataSetPtr dataset, const Config& cfg) = 0;
 };
 
+// returns true if the text of FaissException is about non-recognizing fourcc
+static inline bool
+is_faiss_fourcc_error(const char* what) {
+    if (what == nullptr) {
+        return false;
+    }
+
+    std::string error_msg(what);
+
+    // check if this is fourCC problem
```

Comment:
Using error_msg for judgment can be risky; I suggest using a specific exception instead.